arcgis-ios-sdk-samples/AppDelegate.swift

```diff
@@ -41,6 +41,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UISplitViewControllerDele
         splitViewController.minimumPrimaryColumnWidth = 320
         splitViewController.maximumPrimaryColumnWidth = 320
         
+        // Decode and populate Categories.
+        let contentCollectionViewController = (splitViewController.viewControllers.first as! UINavigationController).viewControllers.first as! ContentCollectionViewController
```

Comment:
I think I'll leave it the way it is. It is fairly common for view controllers to be given a reference to the data they need to display. `ContentTableViewController`, for example, is given an array of samples to display.

Reply:
I agree with @philium here, but I would prefer to see the `[Category]` building code refactored to either a function or property on `AppDelegate` that is used to set the `contentCollectionViewController.categories` property. It would make this code a bit cleaner and it's purpose clearer.

Right now we have:

1. Get the view controller
2. Try a couple of things that could throw exceptions.
3. Now back to the view controller. Let's set its `categories` property to something else that could throw an exception.

Would be much cleaner to refactor that away to:
1. Get the view controller.
2. Set its categories to `AppDelegate.readCategoriesFromConfiguration()`.

And of course `readCategoriesFromConfiguration()` will return a `[Category]` or else will `fatalError()`.

Reply:
``` Swift
lazy var configuredCategories: [Category] = {
    do {
        let url = Bundle.main.url(forResource: "ContentPList", withExtension: "plist")!
        let data = try Data(contentsOf: url)
        return try PropertyListDecoder().decode([Category].self, from: data)
    } catch {
        fatalError("Error decoding categories: \(error)")
    }
}()
```
and
``` Swift
// Decode and populate Categories.
let contentCollectionViewController = (splitViewController.viewControllers.first as! UINavigationController).viewControllers.first as! ContentCollectionViewController
contentCollectionViewController.categories = configuredCategories
```
seems to work nicely.

@philium - what do you think? I think this cleans up the logic a bit. Rest of the PR looks good to me.